src/formatDate.js

```diff
@@ -50,7 +50,38 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let dateArray = date.split(fromFormat[3]);
+  let fromFormatCopy = [...fromFormat];
+  const result = [];
+  let add = 0;
+
+  if (fromFormatCopy[0][0] !== toFormat[0][0]) {
+    fromFormatCopy = fromFormatCopy.reverse();
+    dateArray = dateArray.reverse();
+  };
```

Comment:
that does not work for cases when position 2 switched with position 3
1, 2, 3 => 1, 3, 2

Reply:
![image](https://user-images.githubusercontent.com/35385361/102777820-09995380-439a-11eb-8c14-bc725f005b87.png)